pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1426,6 +1430,20 @@ public LLRealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableCo
     try {
       _partitionGroupConsumerSemaphore.acquire();
```

Comment:
cc @ankitsultana This might be the root cause of #10552. If for some reason 2 segments are going through the OFFLINE -> CONSUMING state transition (one segment just committed), and the newer segment acquired this lock, the older segment will block here and never finish the OFFLINE -> CONSUMING state transition.

Reply:
Yeah that's what I think as well.

I wasn't sure why two OFFLINE -> CONSUMING transitions may happen for two segments of the same partition on the same server so hence wanted to reproduce the issue once. One possible scenario I could think of was: when a server is restarted, the other server might commit a segment (say segment 10) and before the restarted server could process the offline -> consuming transition for segment 10, it might start processing the same for segment 11.